### PR TITLE
API Docs Design

### PR DESCRIPTION
**Pull request by wesrice**
This PR adds a design for the API docs page. It also updates components that are shared between the API Reference template and the other layout templates.

### CODE REVIEW

File: gatsby/src/specs/minfraud/2.0/utils.ts [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+  if (example !== undefined) {
+    return deeplyStripKey(example, '$$ref', (val: unknown) => {
+      // do a couple of quick sanity tests to ensure the value
+      // looks like a $$ref that swagger-client generates.
+      return typeof val === 'string' && val.indexOf('#') > -1;
+    });
+  }
+
+  if (allOf) {
+    const foo = {
```
Comment (kevcenteno) [inline gatsby/src/specs/minfraud/2.0/utils.ts:109]: What's the purpose of foo?

File: gatsby/src/specs/minfraud/2.0/utils.ts [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+    }
+
+    if ( additionalProperties === true ) {
+      (obj as any).additionalProp1 = {
+      };
+    } else if ( additionalProperties ) {
+      const additionalProps = objectify(additionalProperties);
+      const additionalPropVal = sampleFromSchema(additionalProps, config);
+
+      for (let i = 1; i < 4; i++) {
```
Comment (kevcenteno) [inline gatsby/src/specs/minfraud/2.0/utils.ts:149]: Where does 4 come from?

File: gatsby/src/templates/ApiReference/Schema/Content/Content.module.scss [hunk 3]
```diff
@@ -0,0 +1,77 @@
+@import '../styles/variables';
+
+.schema__description {
```
Comment (kevcenteno) [inline gatsby/src/templates/ApiReference/Schema/Content/Content.module.scss:3]: It seems redundant to prefix with `schema__`, when we have the benefit of css modules preprending the module name to the class name. Was there a reason for this BEM-like approach?

File: gatsby/src/components/Layout/Header.module.scss [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+.header {
+  background: var(--mm-color-background);
+  border-bottom: 1px solid var(--mm-color-border);
+  height: var(--page-header-height);
+  position: sticky;
+  top: 0;
+  width: 100%;
+  z-index: 10;
+
+  &::before {
```
[the remark this answers is not in the document]
Comment (wesrice) [reply gatsby/src/components/Layout/Header.module.scss:12]: This is just the 10px line that goes across the top of the page. It's meant to be a branding/design element, signifying the page product type (minfraud vs geoip vs other).

File: gatsby/src/components/CodeSet.tsx [hunk 5]
```diff
@@ -69,49 +70,55 @@ const CodeSet: React.FC = (props) => {
...
+    <div
+      key={key}
+    >
+      {React.Children.map(orderedChildren, child => {
+        if (React.isValidElement(child)) {
+          if (child.props.children.props.className ===
+        activeLanguage) {
+            return (
+              <Pre
+                nav={nav}
```
[the remark this answers is not in the document]
Comment (wesrice) [reply gatsby/src/components/CodeSet.tsx:115]: Yeah, I'd like to improve this too. I needed some wrapping elements around `<Pre />`  and `<CodeSet />` elements, just for styling purposes. However, `<CodeSet />` implements `<Pre />`, so if both `<CodeSet />` and `<Pre />` implemented the wrapping elements, then we end up with nested wrapping elements.

Instead, I think I can add a boolean prop to `<Pre />`, and `<CodeSet />` can signify to the `<Pre />` implementation not to wrap elements for styling.

File: gatsby/src/specs/minfraud/2.0/utils.ts (hunk 1 above)
Comment (wesrice) [reply gatsby/src/specs/minfraud/2.0/utils.ts:109]: Ah, I just forgot to think of a proper name for it. I'll rename it to `composedSchema`. The `OpenApi`  spec allows for Schema composition by using `allOf` and then listing an array of multiple schemas.

File: gatsby/src/specs/minfraud/2.0/utils.ts (hunk 2 above)
Comment (wesrice) [reply gatsby/src/specs/minfraud/2.0/utils.ts:149]: I took this `sampleFromSchema()` function from https://github.com/swagger-api/swagger-ui/blob/074eaa5d51b607bf5293d9519559b133d8295215/src/core/plugins/samples/fn.js#L34, and modified it for our needs.

I'll going to clean this up a bit. We don't have a testing suite in place now, but this is an area that we need to test. I'll add `TODO` comments to remind us.

File: gatsby/src/templates/ApiReference/Schema/Content/Content.module.scss (hunk 3 above)
Comment (wesrice) [reply gatsby/src/templates/ApiReference/Schema/Content/Content.module.scss:3]: I was using BEM conventions before I refactored a really large component into several smaller ones. I'll go through and remove these.

**Comment by wesrice** [comment]
@kevcenteno I think this is ready for another review.

**Pull request merged**